maven/plugins/org.jboss.tools.maven.springboot/src/org/jboss/tools/maven/springboot/configurators/SpringBootProjectConfigurator.java

```diff
@@ -210,6 +212,7 @@ private void configure(IMavenProjectFacade facade) {
 				public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
 					try {
 						MavenProject mavenProject = facade.getMavenProject(monitor);
+						ModuleCoreNature.addModuleCoreNatureIfNecessary(facade.getProject(), monitor);
 						addUtilityFacet(mavenProject, facade.getProject(), monitor);
 						if (ModuleUtils.hasModuleCoreNature(project)) {
```

Comment:
I'd think that this check isn't required any longer. Or if it is we should most likely error (what else than an error is it if we set the core nature and then the check for it 2 lines further down fails?).

Reply:
I was already there and as it was about the root project, I did not want to modify but can remove it if you say so